common/src/main/java/io/netty/util/concurrent/Promise.java

```diff
@@ -41,7 +41,7 @@
      * <p>
      * If it is success or failed already it will throw an {@link IllegalStateException}.
      */
-    Promise<V> setFailure(Throwable cause);
+    Future<V> setFailure(Throwable cause);
```

Comment:
This does not look correct. It should return the promise

Reply:
I didn't see any usages that relied on these returning `Promise` but returning `Future` was convenient. I can change it back, though.

Reply:
Imho it is confusing as it is not clear if it would return a new instance or not ... make it return promise makes it more clear what are the semantics and that your are ok to ignore the return value